src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +43,13 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        OptionalDouble average = IntStream.range(0, numbers.size())
+                .mapToObj(i -> (i % 2 == 1) ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(i -> i % 2 == 1)
+                .mapToInt(i -> i)
+                .average();
```

Comment:
```suggestion
                .average().getAsDouble();
```
при геті автоматично викидується NoSuchElementException якщо опшинал пустий